freqtrade/optimize/space/decimalspace.py

```diff
@@ -1,47 +1,29 @@
-import numpy as np
-from skopt.space import Integer
+from decimal import Decimal
 
+from optuna.distributions import FloatDistribution
 
-class SKDecimal(Integer):
+
+class SKDecimal(FloatDistribution):
     def __init__(
         self,
-        low,
-        high,
-        decimals=3,
-        prior="uniform",
-        base=10,
-        transform=None,
+        low: float,
+        high: float,
+        step: float | None = None,
+        decimals: int = 4,
```

Comment:
Any particular reason we switch from 3 to 4 here? 

I'd keep the default as is ... there's little reason not to i'd think?

Reply:
Was a test tor testing if with 4 all test passes (using decimal localcontext). Changed back to 3 after removing decimal.

Reply:
the only tests failed are:
```
mypy freqtrade
freqtrade/optimize/space/optunaspaces.py:37: error: Incompatible types in assignment (expression has type "int | float", variable has type "int")  [assignment]
freqtrade/optimize/space/optunaspaces.py:38: error: Incompatible types in assignment (expression has type "int | float", variable has type "int")  [assignment]
Found 2 errors in 1 file (checked 315 source files)
```

Reply:
well CI has a different problem :laughing: (optunaspaces is missing completely).
So i can't really look at why we get a mypy error ...

Reply:
yes, sorry, my mistake. I forgot to add optunaspaces.py file. Solved